daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -672,10 +553,11 @@ private[speedy] case class PartialTransaction(
     */
   def beginTry: PartialTransaction = {
     val nid = NodeId(nextNodeIdx)
-    val info = TryContextInfo(nid, context, activeState, authorizers = context.info.authorizers)
+    val info = TryContextInfo(nid, context, authorizers = context.info.authorizers)
     copy(
       nextNodeIdx = nextNodeIdx + 1,
       context = Context(info).copy(nextActionChildIdx = context.nextActionChildIdx),
+      keysState = keysState.beginRollback(),
```

Comment:
Key state really seems to be about more than keys: You include the full consumption info in there which we use for activeness checks even if keys are not involved at all. Maybe ContractState would be a better name? I don’t feel strongly about this so happy to stick with KeyState as well.